include/libtorrent/hasher512.hpp

```diff
@@ -112,7 +117,7 @@ namespace libtorrent
 		gcry_md_hd_t m_context;
 #elif TORRENT_USE_COMMONCRYPTO
 		CC_SHA512_CTX m_context;
-#elif TORRENT_USE_CRYPTOAPI
+#elif TORRENT_USE_CRYPTOAPI && defined(CALG_SHA_512)
```

Comment:
XP pre SP3 does not support only SHA512, so other cryptoapi functions should work well. To detect CALG_SHA_512 absence TORRENT_USE_CRYPTOAPI must be defined and wincrypt.h must be included, so config.hpp is not suitable place for this detection.